testdata/primitives.txt

```diff
@@ -35,24 +35,24 @@ godump_test.Node {
    Bool2Ptr: &false,
    StringPtr: &"foo bar",
    UintptrPtr: &0x499602d2,
-   TypedInt: 123,
-   TypedInt8: -45,
-   TypedInt16: 6789,
-   TypedInt32: -987,
-   TypedInt64: 3849876543247876432,
-   TypedUint: 837,
-   TypedUint8: 38,
-   TypedUint16: 3847,
-   TypedUint32: 9843,
-   TypedUint64: 2834,
-   TypedFloat32: 123.475,
-   TypedFloat64: -12345.09876,
-   TypedComplex64: (0+12.987i),
-   TypedComplex128: (0-473i),
-   TypedBool1: true,
-   TypedBool2: false,
-   TypedString: "foo bar",
-   TypedUintptr: 0x499602d2,
+   TypedInt: godump_test.IntType(123),
```

Comment:
Why would someone be interested in this format?

I mean the previous one was fine

Reply:
Well, as mentioned in the PR description. We use this library in our company in tests failure messages.

The tests basically compare two data structures (large ones) that have so many distinct named primitives. On failure, we print those two values using `Dumper`.

So in our case, it's important to include the types because we need to know which part is different. Because sometimes two values are equal but have different types. When the test fails and we look at the error log. They look the same. So we need to manually investigate each of those values manually (which makes the logs useless).

I hope this makes sense. I agree with you that some users don't want to see those types. So, do you think we should add a toggle to the `Dumper` like:

```go
dumper.ShowPrimitiveTypes = true // or similar name.
``` 

Because again. `godump` is not a formatting library, it's a dumper that describes how a variable look like.

Reply:
I think a setting would be useful, yes.

I'm unsure what should be the default value, but at least people will be able to choose

Reply:
Ok thanks.

However, regarding your last sentence. I'm unsure too. Because currently we're assuming that  `ShowPrimitiveTypes` will enable types for named primitives. But what if later we decided to allow users to enable unnamed types too (string, int8, float32, ...). In that case we would need to add another option and will end up with  two options for just primitive types.

I wonder if we can use an enum:

```go
...

dumper.ShowPrimitiveTypes = godump.PT_ALL
// or
dumper.ShowPrimitiveTypes = godump.PT_NAMED
// or
dumper.ShowPrimitiveTypes = 0 // disable primitive types (default)
```

Reply:
Indeed it might be better